### PR TITLE
fix #50

### LINKED ISSUE #50: Pontential memory leak in worker mode
When PHP is compiled with the `--enable-debug` flag, a potential memory leak is reported in worker mode when running the tests:

```
go test -v

[Fri Oct 21 08:21:09 2022]  Script:  '-'
/Users/dunglas/workspace/php-src/Zend/zend_hash.c(279) :  Freeing 0x000000012aa5e000 (56 bytes), script=-
Last leak repeated 3 times
=== Total 4 memory leaks detected ===
```

I tried to track the potential leak with Valgrind but didn't find anything obvious yet.

### PR DESCRIPTION
**Pull request by krakjoe**

### CODE REVIEW

**Comment by krakjoe** [comment]
So, I'm not sure why hash_environment is being called in a shutdown handler, it looks strange and the comment was missing a reference link to a test or any code.

When you watch the addresses leaking, this is what appears to be missing, you either need to find the case where worker_request_shutdown is called, or destroy symbols if they exist ... I chose the latter in the end because it looks clearer to me ...

I may be missing context ... hopefully this is a nudge in the right direction, even if not the final solution :)

**Comment by dunglas** [comment]
Thank you very much for your help and contribution!

> So, I'm not sure why hash_environment is being called in a shutdown handler, it looks strange and the comment was missing a reference link to a test or any code.

This was to prevent a segfault, but if tests are green, that means it's probably no longer necessary thanks to your fix.

Before your patch, we were destroying superglobals after each request processed in the worker script. But `php_request_shutdown()` does this too, so it was necessary to re-create superglobals when the script was terminating (before calling `php_request_shutdown()`) to prevent the segfault. It looks like your patch somehow prevents this segfault too.

> I may be missing context ...

Basically, this code is related to the worker mode. The main idea is to keep the same script running (with all its variables, objects), etc, but handle multiple HTTP requests with it. So everything is kept in memory, but each time `frankenphp_handle_request()` is called, context-related elements (superglobals and `php://input`...) are re-created with data coming from the new requests, and destroyed when the response has been generated.

I still don't have a complete understanding of the PHP code base, and I thought that `EG(symbol_table)` contained all the symbols. Not just the superglobals. Is there any documentation (or book) available to better understand these parts?

Thank you again!!

**Comment by krakjoe** [comment]
@dunglas now the code reflects requirements as described ... I'm a bit more confident that this is correct, what do you think ?

**Review by dunglas — APPROVED** [review]
Indeed, this looks good to me!

**Comment by dunglas** [comment]
Thank you @krakjoe!

**Comment by dunglas** [comment]
This change triggers this error on Mac (not tested yet locally on Linux):

```
Assertion failed: (zval_gc_type((ref)->gc.u.type_info) == 7 || zval_gc_type((ref)->gc.u.type_info) == 8), function gc_possible_root, file zend_gc.c, line 647.
SIGABRT: abort
PC=0x187906d98 m=18 sigcode=0
signal arrived during cgo execution


goroutine 146 [syscall, locked to thread]:
runtime.cgocall(0x102edfb28, 0x14000139d48)
	/opt/homebrew/Cellar/go/1.19.2/libexec/src/runtime/cgocall.go:158 +0x54 fp=0x14000139d10 sp=0x14000139cd0 pc=0x102c10de4
github.com/dunglas/frankenphp._Cfunc_frankenphp_request_shutdown()
	_cgo_gotypes.go:783 +0x38 fp=0x14000139d40 sp=0x14000139d10 pc=0x102e99ec8
github.com/dunglas/frankenphp.go_execute_script(0x14000481d01?)
	/Users/dunglas/workspace/frankenphp/frankenphp.go:394 +0x1f8 fp=0x14000139de0 sp=0x14000139d40 pc=0x102e9b4d8
_cgoexp_395c8cbf2c19_go_execute_script(0x103f0f5c0?)
	_cgo_gotypes.go:887 +0x20 fp=0x14000139e00 sp=0x14000139de0 pc=0x102e9e9f0
runtime.cgocallbackg1(0x102e9e9d0, 0x0?, 0x0)
	/opt/homebrew/Cellar/go/1.19.2/libexec/src/runtime/cgocall.go:316 +0x244 fp=0x14000139ef0 sp=0x14000139e00 pc=0x102c11204
runtime.cgocallbackg(0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.19.2/libexec/src/runtime/cgocall.go:235 +0xd8 fp=0x14000139f80 sp=0x14000139ef0 pc=0x102c10f38
runtime.cgocallbackg(0x102e9e9d0, 0x16d87eef8, 0x0)
	<autogenerated>:1 +0x1c fp=0x14000139fb0 sp=0x14000139f80 pc=0x102c780cc
runtime.cgocallback(0x0, 0x0, 0x0)
	/opt/homebrew/Cellar/go/1.19.2/libexec/src/runtime/asm_arm64.s:1094 +0xa0 fp=0x14000139fe0 sp=0x14000139fb0 pc=0x102c76a50
runtime.goexit()
	/opt/homebrew/Cellar/go/1.19.2/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000139fe0 sp=0x14000139fe0 pc=0x102c76b24
```

I wonder if it's not related to the problem I was explaining here: https://github.com/dunglas/frankenphp/pull/76#issuecomment-1297704220

**Comment by dunglas** [comment]
This works on Linux, so it's a Mac-only issue.
